homeassistant/components/ohme/coordinator.py

```diff
@@ -3,69 +3,72 @@
 from datetime import timedelta
 import logging
 
-from ohme import ApiException, AuthException, OhmeApiClient
+from ohme import ApiException, OhmeApiClient
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import ConfigEntryError
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-type OhmeConfigEntry = ConfigEntry[OhmeCoordinator]
+type OhmeConfigEntry = ConfigEntry[list[DataUpdateCoordinator]]
 
 
-class OhmeCoordinator(DataUpdateCoordinator[None]):
+class OhmeChargeSessionCoordinator(DataUpdateCoordinator[None]):
```

Comment:
I would maybe recommend creating a base Coordinator. This way you can remove the constructor from the subcoordinators and you can automatically wrap all update statements in the same try catch so you don't have to duplicate it.

For example in Withings I have
```py


    async def _async_update_data(self) -> _DataT:
        try:
            return await self._internal_update_data()
        except (WithingsUnauthorizedError, WithingsAuthenticationFailedError) as exc:
            raise ConfigEntryAuthFailed from exc

    @abstractmethod
    async def _internal_update_data(self) -> _DataT:
        """Update coordinator data."""
```

(In your case you don't have different data types since both coordinators have the `None` type)

Reply:
One of my coordinators then looks like this
```py

class WithingsDeviceDataUpdateCoordinator(
    WithingsDataUpdateCoordinator[dict[str, Device]]
):
    """Withings device coordinator."""

    coordinator_name: str = "device"
    _default_update_interval = timedelta(hours=1)

    async def _internal_update_data(self) -> dict[str, Device]:
        """Update coordinator data."""
        devices = await self._client.get_devices()
        return {device.device_id: device for device in devices}
```
Which is quite small and neat and without a lot of code duplication

Reply:
You beat me to it :) Just started to do the same thing. That _internal_update_data idea saves a bit more duplication so I'll put that in too